ui/src/app/applications/components/application-retry-options/application-retry-options.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import {FormApi, NestedForm, Text, Form} from 'react-form';
+import {Checkbox, FormField} from 'argo-ui';
+import {NumberField} from '../../../shared/components';
+import * as models from '../../../shared/models';
+
+require('./application-retry-options.scss');
+
+function buildFormItem(label: string, propertyPath: string, component: React.ComponentType, formApi: FormApi, componentProps?: Record<string, any>) {
+    return <FormField formApi={formApi} label={label} field={propertyPath} component={component} componentProps={componentProps} />;
+}
+
+const onlyPositiveValidation = {
+    min: '1',
+    step: '1'
+};
+
+const retryOptions: Array<(formApi: FormApi) => React.ReactNode> = [
+    formApi => buildFormItem('Limit', 'limit', NumberField, formApi, onlyPositiveValidation),
+    formApi => buildFormItem('Duration', 'backoff.duration', Text, formApi),
+    formApi => buildFormItem('Max Duration', 'backoff.maxDuration', Text, formApi),
+    formApi => buildFormItem('Factor', 'backoff.factor', NumberField, formApi, onlyPositiveValidation)
+];
+
+const durationRegex = /\b((\d+(\.\d+)?)\s*(d))?(\s*(\d+)\s*(h|hr|hrs?|hours?))?(\s*(\d+)\s*(m|min|mins?|minutes?))\b/m;
```

Comment:
Backend parses duration using https://pkg.go.dev/time#ParseDuration. It does not support days but supports seconds. Please change allow only following time units: `s`, `m`, `h` (so no days and single character units only)